### PR TITLE
FSET-1632 Progress non-siftable candidates to assessment centre

### PR DESCRIPTION
**Pull request by benjstephenson**

### CODE REVIEW

**Comment by andrewrembrandt** [comment]
Tests all pass locally

File: app/model/command/ApplicationForFsac.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+case class ApplicationForFsac(
+  applicationId: String,
+  evaluationResult: PassmarkEvaluation
+)
+
+object ApplicationForFsac{
+  implicit val applicationForFsacFormat: OFormat[ApplicationForFsac] = Json.format[ApplicationForFsac]
+
+  implicit def applicationForFsacBsonReads(document: BSONDocument): ApplicationForFsac = {
```
Comment (kjharkin) [inline app/model/command/ApplicationForFsac.scala]: This class should have no knowledge of repositories or mongo. It's a model class - please refactor this code.

Comment (benjstephenson) [inline app/model/command/ApplicationForFsac.scala]: We put the bson read implicits in other classes that are returned from repositories, this one just requires a manual read as it doesn't match the doc structure.

File: app/scheduler/ProgressToAssessmentCentreJob.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      case applications => assessmentCentreService.progressApplicationToAssessmentCentre(applications).map { result =>
+        play.api.Logger.info(
+          s"Progress to assessment centre complete - ${result.successes.size} updated and ${result.failures.size} failed to update"
+        )
+      }
+    }
+  }
+}
+
+object TbcConfig extends BasicJobConfig[WaitingScheduledJobConfig](
```
Comment (kjharkin) [inline app/scheduler/ProgressToAssessmentCentreJob.scala]: More descriptive name?

File: app/services/assessmentcentre/AssessmentCentreService.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  val applicationRepo = repositories.applicationRepository
+  val assessmentCentreRepo = repositories.assessmentCentreRepository
+}
+
+trait AssessmentCentreService {
+
+  def applicationRepo: GeneralApplicationRepository
+  def assessmentCentreRepo: AssessmentCentreRepository
+
+  def nextApplicationForAssessmentCentre(batchSize: Int): Future[Seq[ApplicationForFsac]] = {
```
Comment (kjharkin) [inline app/services/assessmentcentre/AssessmentCentreService.scala]: Method should should be plural?

File: app/services/assessmentcentre/AssessmentCentreService.scala [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    assessmentCentreRepo.nextApplicationForAssessmentCentre(batchSize)
+  }
+
+  def progressApplicationToAssessmentCentre(applications: Seq[ApplicationForFsac]): Future[SerialUpdateResult[ApplicationForFsac]] = {
```
Comment (kjharkin) [inline app/services/assessmentcentre/AssessmentCentreService.scala]: Method name should be plural?

File: test/services/assessmentcentre/AssessmentCentreServiceSpec.scala [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import repositories.application.GeneralApplicationRepository
+import repositories.assessmentcentre.AssessmentCentreRepository
+import testkit.{ ExtendedTimeout, FutureHelper }
+
+import scala.concurrent.Future
+
+class AssessmentCentreServiceSpec extends PlaySpec with OneAppPerSuite with Results with ScalaFutures with FutureHelper with MockFactory
+  with ExtendedTimeout {
+  "An AssessmentCentreService" should {
+    case class TestAssessmentCentreService(acRepo: AssessmentCentreRepository, appRepo: GeneralApplicationRepository)
```
Comment (kjharkin) [inline test/services/assessmentcentre/AssessmentCentreServiceSpec.scala]: Why is this a case class?

File: app/model/command/ApplicationForFsac.scala (hunk 1 above)
Comment (kjharkin) [inline app/model/command/ApplicationForFsac.scala]: Previous comment stands - wrong class/package

Comment (benjstephenson) [inline app/model/command/ApplicationForFsac.scala]: I disagree but I'll move it somewhere else.

File: test/services/assessmentcentre/AssessmentCentreServiceSpec.scala (hunk 5 above)
Comment (benjstephenson) [inline test/services/assessmentcentre/AssessmentCentreServiceSpec.scala]: Don't know, will rework.

File: app/scheduler/ProgressToAssessmentCentreJob.scala (hunk 2 above)
Comment (andrewrembrandt) [inline app/scheduler/ProgressToAssessmentCentreJob.scala]: This shouldn't have made it in the branch

File: test/services/assessmentcentre/AssessmentCentreServiceSpec.scala (hunk 5 above)
Comment (andrewrembrandt) [inline test/services/assessmentcentre/AssessmentCentreServiceSpec.scala]: There was a reason, but no longer required

**Pull request merged**